test/controllers/PaymentControllerSpec.scala

```diff
@@ -16,157 +16,98 @@
-      }
-    }
-
-    "an SA UTR is missing from the user" in new SetupTestPaymentController(isAgent = true, Future.successful(PaymentJourneyModel("id", "redirect-url"))) {
-      setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess)
-
-      val result: Future[Result] = testController.agentPaymentHandoff(testAmountInPence)(
-        fakeRequestWithActiveSession.withSession(sessionUtils.SessionKeys.clientMTDID -> "XAIT00000000015"))
-      status(result) shouldBe SEE_OTHER
-      redirectLocation(result) shouldBe Some(controllers.agent.routes.EnterClientsUTRController.show.url)
-    }
-
-    "return an error" when {
-
-      "an error response is returned by the connector" in new SetupTestPaymentController(isAgent = true, Future.successful(PaymentJourneyErrorResponse(
-        INTERNAL_SERVER_ERROR, "Error Message"))) {
-        setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess)
-
-        val result: Future[Result] = testController.agentPaymentHandoff(testAmountInPence)(fakeRequestConfirmedClient())
-
-        result.failed.futureValue shouldBe an[Exception]
-      }
-
-      "an exception is returned by the connector" in new SetupTestPaymentController(isAgent = true, Future.failed(new Exception("Exception Message"))) {
-        setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess)
-
-        val result: Future[Result] = testController.agentPaymentHandoff(testAmountInPence)(fakeRequestConfirmedClient())
-
-        result.failed.futureValue shouldBe an[Exception]
-        result.failed.futureValue.getMessage shouldBe "Exception Message"
-
+class PaymentControllerSpec extends MockAuthActions {
+
+  lazy val mockPayApiConnector: PayApiConnector = mock(classOf[PayApiConnector])
+
+  override def fakeApplication(): Application = applicationBuilderWithAuthBindings()
+    .overrides(
+      api.inject.bind[PayApiConnector].toInstance(mockPayApiConnector)
+    ).build()
+
+  val testController = fakeApplication().injector.instanceOf[PaymentController]
+
+  val paymentJourneyModel = PaymentJourneyModel("id", "redirect-url")
+
+
+  mtdAllRoles.foreach { case mtdUserRole =>
+    val isAgent = mtdUserRole != MTDIndividual
+    val action = if (isAgent) testController.agentPaymentHandoff(testAmountInPence) else testController.paymentHandoff(testAmountInPence)
+    val fakeRequest = fakeGetRequestBasedOnMTDUserType(mtdUserRole)
+    s"${if (isAgent) "agentP" else "p"}aymentHandoff" when {
+      s"the $mtdUserRole is authenticated" should {
+        if (mtdUserRole == MTDSupportingAgent) {
+          testSupportingAgentDeniedAccess(action)(fakeRequest)
+        } else {
+          "redirect to the payment api provided by payAPI" when {
+            "a successful payments journey is started" in {
+              setupMockSuccess(mtdUserRole)
+              mockSingleBusinessIncomeSource()
+              when(mockPayApiConnector.startPaymentJourney(ArgumentMatchers.eq(testSaUtrId), ArgumentMatchers.eq(BigDecimal(10000)),
```

Comment:
not sure if this is the way calling API / service consuming this amount, but I would prefer to provide amount in pounds and pence, ie: BigDecimal(10000.19)